lib/common/client/concerns/session_based_client.rb

```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+module Common
+  module SessionBasedClient
+    extend ActiveSupport::Concern
+
+    attr_reader :config, :session
+
+    def authenticate
+      binding.pry
+      if session.expired?
+        session = get_session
+        session.save
+      end
+      self
+    end
+
+    private
+
+    def auth_headers
+      config.base_request_headers.merge('appToken' => config.app_token, 'mhvCorrelationId' => session.user_id.to_s)
```

Comment:
But to your point, I'm still trying to work on making Client and Configuration more composable and easy to use. That's the goal at least.